app/controllers/users_controller.rb

```diff
@@ -16,9 +16,11 @@ def show
   def finish_signup
     @user = User.find(params[:id])
 
+    # mehr rails like wäre eine Trennung der Actions nach HTTP-Methode. Die zusätzliche Prüfung im if nach den Parametern ist zumindest ungewöhnlich, weil ja die Validierung einen fehlenden Parameter schon abfangen würde.
```

Comment:
Additional parameters are checked here because they're only need for adding multiple oAuth strategies to one account. It's however totally fine to just create a new user with one and the same route.

I don't see the immediate benefit of using two methods for GET and PATCH here.